src/dmd/backend/machobj.d

```diff
@@ -1785,6 +1785,12 @@ int Obj_comdat(Symbol *s)
         flags = S_COALESCED | S_ATTR_PURE_INSTRUCTIONS | S_ATTR_SOME_INSTRUCTIONS;
         s.Sseg = getsegment2(seg_textcoal_nt, sectname, segname, align_, flags);
     }
+    else if ((s.ty() & mTYLINK) == mTYweakLinkage)
```

Comment:
I'm not sure if this is the best solution. What I need is a symbol with weak linkage, stored in a specific section. DMD seems to implement weak linkage using the `SCcomdat ` storage class: https://github.com/dlang/dmd/blob/5fda4ea088f70afb99818482e8482771758136a2/src/dmd/backend/cdef.d#L912
The problem with that is that it will hardcode a section for the symbol.